vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -318,25 +375,49 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 		for _, mount := range containerJSONInfo.Mounts {
 			// check if the mount location belongs to vsphere plugin
 			if matchNameforVMDK(mount.Source) {
-				r.Incr(mount.Name)
-				log.Debugf("  name=%v (driver=%s source=%s) (%v)",
+				// populate temp map
+				rc := auxRefMap[mount.Name]
+				if rc == nil {
+					rc = newRefCount()
+					auxRefMap[mount.Name] = rc
+				}
+				rc.count++
+
+				log.Infof("  name=%v (driver=%s source=%s) (%v)",
 					mount.Name, mount.Driver, mount.Source, mount)
 			}
 		}
 	}
 
-	// Check that refcounts and actual mount info from Linux match
-	// If they don't, unmount unneeded stuff, or yell if something is
-	// not mounted but should be (it's error. we should not get there)
+	// check if the background refcount was dirtied
+	r.dirtyMtx.Lock()
+	if r.isDirty == false {
+		// if not, copy auxRefMap into refMap and unmount unwanted volumes
```

Comment:
Why even bother to copy if we can just ignore (and we already do ignore) the map when dirty ?  What am I missing ?